janus_collector/src/lib.rs

```diff
@@ -0,0 +1,1046 @@
+//! DAP protocol collector
+
+use backoff::{backoff::Backoff, ExponentialBackoff};
+use derivative::Derivative;
+use janus_core::{
+    hpke::{self, associated_data_for_aggregate_share, HpkeApplicationInfo, HpkePrivateKey},
+    message::{CollectReq, CollectResp, HpkeConfig, Interval, Role, TaskId},
+    retries::{http_request_exponential_backoff, retry_http_request},
+    task::{url_ensure_trailing_slash, AuthenticationToken},
+};
+use prio::{
+    codec::{Decode, Encode},
+    vdaf,
+};
+use reqwest::{
+    header::{HeaderValue, ToStrError, CONTENT_TYPE, LOCATION, RETRY_AFTER},
+    StatusCode,
+};
+use retry_after::{FromHeaderValueError, RetryAfter};
+use std::{
+    convert::TryFrom,
+    time::{Duration as StdDuration, SystemTime},
+};
+use tokio::time::{sleep, Instant};
+use url::Url;
+
+/// HTTP header where auth tokens are provided in messages to the leader.
+pub const DAP_AUTH_HEADER: &str = "DAP-Auth-Token";
```

Comment:
nit: this constant could be exported from `janus_core` but this business should get replaced with a more robust authentication story before too long anyway, so it's not a big deal.